models/components/versions/base/component.go

```diff
@@ -0,0 +1,22 @@
+package base
+
+import (
+	"github.com/opencontrol/compliance-masonry/models/common"
+)
+
+type Component interface {
+	GetName() string
+	GetKey() string
+	SetKey(string)
+	GetAllSatisfies() []Satisfies
+	GetVerifications() *common.VerificationReferences
+	GetReferences() *common.GeneralReferences
+	GetVersion() float32
+}
+
+type Satisfies interface {
+	GetStandardKey() string
+	GetControlKey() string
+	GetNarrative() string
+	GetCoveredBy() common.CoveredByList
+}
```

Comment:
This probably warrants some developer-centric documentation (or at least a comment at the top of the file saying "we are using `interface`s to provide an abstraction between schema versions. See `some/path/` for the implementations of the different versions."